packages/client/src/app/jobs/executeJobModal.tsx

```diff
@@ -1,17 +1,30 @@
+import React, { useState, useMemo, useCallback } from 'react';
 import { Button, Modal, Table, Text } from '@nextui-org/react';
 import { MitmControlDTO } from '@rotom/connections';
 import { Selection } from '@react-types/shared/src/selection';
 import { toast } from 'react-toastify';
-import { useCallback, useState } from 'react';
+
+import { SearchInput } from '../status/search';
 
 interface ExecuteJobModalProps {
   closeModal: () => void;
   devices?: MitmControlDTO[];
   jobId: string;
 }
 
-export const ExecuteJobModal = ({ closeModal, devices, jobId }: ExecuteJobModalProps): JSX.Element => {
+export const ExecuteJobModal: React.FC<ExecuteJobModalProps> = ({ closeModal, devices, jobId }) => {
   const [selectedDevices, setSelectedDevices] = useState<Selection>();
+  const [search, setSearch] = useState('');
+
+  const filteredDevices = useMemo(() => {
```

Comment:
@na-ji I am not the best in React, but do we always build filteredDevices,even the search is empty?

Reply:
Here, the hook useMemo prevent this. Basically, when doing the first render, it will compute the filtered list even if the search query is empty. On subsequent renders, it will only compute this array again if the search was updated.